### ISSUE TITLE
unable ssl

### ISSUE THREAD

**Issue by Herlderlord** [docker-library/mysql#339]
Hello, 

I'm using the v5.7 of mysql thanks to your docker image and i've an issue. 
I'm trying to configure ssl like this way described here: 
https://dev.mysql.com/doc/refman/5.7/en/using-encrypted-connections.html

So, I've added these lines to the 'my.cnf' file: 
```
ssl-ca=/encrypted/ca-cert.pem
ssl-cert=/encrypted/server-cert.pem
ssl-key=/encrypted/server-key.pem
```
Files generated thanks to these commands: (volume mounted in docker container)
```
openssl genrsa 2048 > ca-key.pem 
openssl req -new -x509 -nodes -days 1000 -key ca-key.pem > ca-cert.pem
openssl req -newkey rsa:2048 -days 1000 -nodes -keyout server-key.pem > server-req.pem
openssl x509 -req -in server-req.pem -days 1000 -CA ca-cert.pem -CAkey ca-key.pem -set_serial 01 > server-cert.pem
openssl req -newkey rsa:2048 -days 1000 -nodes -keyout client-key.pem > client.req.pem
openssl x509 -req -in client-req.pem -days 1000 -CA ca-cert.pem -CAkey ca-key.pem -set_serial 01 > client-cert.pem
```

Problem is that it doesn't work. 
I can show you these two instruction's results : 
```
show status like 'Ssl_version'; 
```
Ssl_version => [nothing]

```
SHOW VARIABLES like '%ssl%'
```
have_openssl => DISABLED
have_ssl => DISABLED

How can i unable ssl (TLS) ? 

Best regards

**Comment by ltangvald** [comment]
Hi,

If you run docker logs <container>, you will likely see:
2017-10-31T09:47:01.948682Z 0 [ERROR] SSL error: Unable to get private key from '/encrypted/server-key.pem'

I think you're probably missing this command (which basically removes the passphrase from the key)
openssl rsa -in server-key.pem -out server-key.pem

**Comment by Herlderlord** [comment]
I've tried to exec your command and then restart my container. 
So, here the errors in the log file:
```
[Note] Skipping generation of SSL certificates as options related to SSL are specified. 
[Warning] Failed to set up SSL because of the following SSL library error: SSL_CTX_set_default_verify_paths failed 
``` 
I'm sorry, i'm a begginer with mysql and there's a lot of little things like it that i'm not used to meet.

**Comment by ltangvald** [comment]
Hm, I followed your steps (but fixed a typo in one command that specified client.cert.pem instead of client-cert.pem).
Which platform are you running Docker on?

**Comment by ltangvald** [comment]
Also, by restarting do you mean you ran "docker restart <container>"? I simply started a new one "docker run"

**Comment by Herlderlord** [comment]
Ubuntu 14.04 with Docker 1.11.2 (i know it's not updated)
By restarting i mean remove the container and then run it again.

**Comment by ltangvald** [comment]
How do you run the image, specifically?

**Comment by Herlderlord** [comment]
This way : 

```
#!/bin/bash

sudo docker build -t test/mysql-server mysql_docker

sudo docker run --name mysqlgafes -e MYSQL_ROOT_PASSWORD=1234 -v `pwd`/mysql_docker/data:/var/lib/mysql -v `pwd`/mysql_docker/certificates/:/encrypted/ -e MYSQL_ROOT_HOST=% test/mysql-server
```

**Comment by ltangvald** [comment]
Hm, thought maybe it could be an issue with having the existing database, but that also works for me (starting with the incorrect setup first, then fixing and restarting).

Do you make any changes in the image you build, i.e. in Dockerfile?

**Comment by ltangvald** [comment]
What permissions do you have on certificates/ and the files inside it?

**Comment by Herlderlord** [comment]
I've only put my files into my container.
Here the permissions: 
-rw-r--r-- 1 mysql  mysql     0 Oct 31 10:02 ca-cert.pem
-rw-r--r-- 1 mysql  mysql  1675 Oct 31 10:02 ca-key.pem
-rw-r--r-- 1 mysql  mysql  1099 Oct 31 09:53 client-cert.pem
-rw-r--r-- 1 mysql  mysql  1704 Oct 31 09:53 client-key.pem
-rw-r--r-- 1 mysql  mysql   956 Oct 31 09:53 client-req.pem
-rw-r--r-- 1 mysql  mysql  1099 Oct 31 09:53 server-cert.pem
-rw-r--r-- 1 mysql  mysql  1675 Oct 31 11:27 server-key.pem
-rw-r--r-- 1 mysql  mysql   956 Oct 31 09:53 server-req.pem

**Comment by tianon** [comment]
Since we can't reproduce (and I'm able to successfully run this image with SSL by supplying `--ssl` and letting the entrypoint auto-generate appropriate certificates), I'm going to close.  This sounds like there's either something going on with your custom `Dockerfile`, permissions, or your local environment.

**Comment by morganjones-stratoscale** [comment]
I see the issue as reported.  Certificates which are imported into the container fail to "openssl verify" whereas the certificates generated inside the container work fine.  The ones generated outside do verify outside.  I have verified via sum(1) that the contents of the certificates are indentical inside and out.

Herlderlord, did you ever find a solution?

**Comment by Herlderlord** [comment]
I don't use my own certificate and let mysql create its own if i remember well.

**Comment by vinaysshenoyhl** [comment]
I have this problem as well. If I use `--ssl-mode=REQUIRED`, it works fine. But either `VERIFY_CA` or `VERIFY_IDENTITY` fails with `ERROR 2026 (HY000): SSL connection error: SSL_CTX_set_default_verify_paths failed`.

However, I do not have a luxury of letting mysql create its own certificates. I use the image for local development(without ssl) and for connecting to an AWS RDS Aurora instance, which has their own key installed.

**Comment by morganjones-stratoscale** [comment]
To follow up:

I ultimately had two issues.  The first was that I wasn't tranforming my keys via "openssl rsa -in -out" - I had seen that you needed to do that if your keys didn't say "RSA key" or some such, but mine already did - turns out I needed to transform them anyways.  My second issue was that I was using the same CN between the two certificates.  With those issues fixed, I was able to get certificates working.
